art/defences/preprocessor/mp3_compression.py

```diff
@@ -0,0 +1,153 @@
+# MIT License
+#
+# Copyright (C) IBM Corporation 2020
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements the MP3 compression defence `Mp3Compression`.
+
+| Paper link: https://arxiv.org/abs/1801.01944
+
+| Please keep in mind the limitations of defences. For details on how to evaluate classifier security in general,
+    see https://arxiv.org/abs/1902.06705  .
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import logging
+from io import BytesIO
+
+import numpy as np
+
+from art.defences.preprocessor.preprocessor import Preprocessor
+
+logger = logging.getLogger(__name__)
+
+
+class Mp3Compression(Preprocessor):
+    """
+    Implement the MP3 compression defense approach.
+
+    """
+
+    params = ["channel_index", "sample_rate"]
+
+    def __init__(self, channel_index, sample_rate, apply_fit=True, apply_predict=False):
+        """Create an instance of MP3 compression.
+
```

Comment:
Could you please give a description for sample_index and sample_rate?

Reply:
Great spot! Have added the missing description :+1: